services/src/main/java/org/keycloak/keys/loader/ClientPublicKeyLoader.java

```diff
@@ -112,16 +113,18 @@ private static KeyWrapper getSignatureValidationKey(CertificateRepresentation ce
             kid = certInfo.getKid() != null ? certInfo.getKid() : KeyUtils.createKeyId(clientCert.getPublicKey());
             keyWrapper.setKid(kid);
             keyWrapper.setPublicKey(clientCert.getPublicKey());
-            keyWrapper.setType(clientCert.getPublicKey().getAlgorithm());
+            keyWrapper.setType(JavaAlgorithm.getKeyType(clientCert.getPublicKey().getAlgorithm()));
             keyWrapper.setCertificate(clientCert);
             keyWrapper.setIsDefaultClientCertificate(true);
+            keyWrapper.setCurve(clientCert.getPublicKey().getAlgorithm());
         } else {
             PublicKey publicKey = KeycloakModelUtils.getPublicKey(encodedPublicKey);
             // Check if we have kid in DB, generate otherwise
             kid = certInfo.getKid() != null ? certInfo.getKid() : KeyUtils.createKeyId(publicKey);
             keyWrapper.setKid(kid);
             keyWrapper.setPublicKey(publicKey);
-            keyWrapper.setType(publicKey.getAlgorithm());
+            keyWrapper.setType(JavaAlgorithm.getKeyType(publicKey.getAlgorithm()));
+            keyWrapper.setCurve(publicKey.getAlgorithm());
```

Comment:
It seems that `setCurve` makes sense if the key is regarding ECDSA or EdDSA.
In case of the key regarding RSA-SSA (RS, PS), it sets the value "RSA". If it does not cause a side-effect, ther is no problem. If causes, need to do some appropriate processing.

Reply:
Done!